### ISSUE TITLE
2_1_3.php bug(s)

### ISSUE THREAD

**Issue by torvista** [mc12345678/zen-colorbox#15]
1) On first run/Admin refresh:
PHP Fatal error:  'break' not in the 'loop' or 'switch' context in ....\installers\zen_colorbox\2_1_3.php on line 53.
I commented out the break and it ran.

2) On first ok run
Message stack says:
"Language file for Zen Colorbox v2.1.2 is not accessible." but in green (so constants have not been deleted)
a) it seems it is not finding the new constant on first pass. I have it only in catalog.... /english, not a template override.
b) it is using the same $current_version throughout this file when it should use 2.1.2 for the previous two executes, and 2.1.3 for the language check.

Messagestack also says "Installed Zen Colorbox v2.1.3", but it has not deleted the constants.

3)  this clause:

> && zen_get_configuration_key_value('ZEN_COLORBOX_SLIDESHOW_START_TEXT') === 'start slideshow')

will fail if someone has changed the text from the original installation, no?

**Comment by mc12345678** [comment]
1. Hadn't used/tried it before that I could remember; however, was trying to treat the included file as being in scope with the loop and therefore incorrectly thinking that the break would break the loop.

Incorporated a modification to work around that and still accomplish desired goal.

2. Hadn't loaded the language file into scope for admin so yes that portion of the situation expected.  Incorporated same load "sequence" into the installer file as used on the catalog side which supports override, base and english (language provided) loading. Also added an internal
variable for version information as didn't see the need to modify the installer to support.

3. Yes that is correct, the database entry is to remain present to ensure that the non-default value remains available to incorporate into the language file if it hasn't already been. R\hen to offer guidance on how to apply a setting so that values would be electronically auto-removed.

**Comment by torvista** [comment]
That seems to work ok now. The only nitpick is the removal of the db constants references 2.13 when I think it should reference 2.12:
![clipboard01](https://user-images.githubusercontent.com/4391026/35621698-9c8caed0-0685-11e8-94ca-2a98d28ab8f4.gif)

I like to be very explicit with messages and would rephrase them:

Configuration table Slideshow Start Text for ...
to:
Table "configuration", constant "ZEN_COLORBOX_SLIDESHOW_START_TEXT" for ...

And for modified constants
Configuration table Slideshow Stop Text for Zen Colorbox v2.1.3 is still defined in the database and is not currently set to its default value of \'stop slideshow\'. This prevents removal from the database in place of the language defined version.

to

Table "configuration", constant "ZEN_COLORBOX_SLIDESHOW_STOP_TEXT" (Slideshow Stop Text) has NOT been removed as it has been changed from its default value of \'stop slideshow\. Please copy your changed text from the admin->Zen Colorbox configuration page->Slideshow Stop Text to your template language file(s) if required and restore the default text value of \'stop slideshow\ to the admin page "Slideshow Stop Text" to allow the installer to remove the constant. 

And then there is the matter of these messages being hard-coded instead of in a language-file...it never ends!!

**Comment by torvista** [comment]
and also when the installer does not continue due to the admin text not being default, the message ies
"Stopped installation of Zen Colorbox at v2.1.3"
...should be 2.12 or 
Stopped installation of Zen Colorbox v2.1.3

**Comment by mc12345678** [comment]
> The only nitpick is the removal of the db constants references 2.13 when I think it should reference 2.12

So, I was thinking that the version information should relate to which one removed the setting, instead of the last version to which the setting applied.  To accomplish that it looks like the terminology could be reworded.  Interesting in understanding why the/a previous version should be referenced.

The amplification of the messages can be done. As to use of a language file, I had considered it, but also that the "brevity" of what is being done didn't think it warranted to be in another language other than English and to also increase the content of the existing language file.  However, to address both concepts and concerns, will be creating a separate "install" language file that will be independently maintained.  This way, those that wish to remove the installation portion(s) can do so and not necessarily affect operation (although thinking that should then also split out the version checking portion so that if the intention is that just installer be removed that the other checking/operations remain available.)

Saw the additional comments in the forum about the instructions and will be considering their incorporation.

**Comment by torvista** [comment]
> Interesting in understanding why the/a previous version should be referenced.
Depends on the grammar I suppose!
![clipboard01](https://user-images.githubusercontent.com/4391026/35769027-0a20c928-0905-11e8-85ca-3370e5cd7afb.jpg)
The constant was FOR 2.12.

> didn't think it warranted to be in another language
No it doesn't, but you haven't left anything else to pick holes in!
